chain/epoch_manager/src/lib.rs

```diff
@@ -747,13 +747,29 @@ impl EpochManager {
         self.cares_about_shard_in_epoch(epoch_id, account_id, shard_id)
     }
 
+    // `shard_id` always refers to a shard in the current epoch that the next block from `parent_hash` belongs
+    // If shard layout will change next epoch, returns true if it cares about any shard
+    // that `shard_id` will split to
     pub fn cares_about_shard_next_epoch_from_prev_block(
         &mut self,
         parent_hash: &CryptoHash,
         account_id: &AccountId,
         shard_id: ShardId,
     ) -> Result<bool, EpochError> {
         let next_epoch_id = self.get_next_epoch_id_from_prev_block(parent_hash)?;
+        if let Some(split_shards) =
+            self.get_split_shards_if_shards_will_change(parent_hash, vec![shard_id])?
+        {
+            for next_shard_id in split_shards.get(&shard_id).unwrap() {
```

Comment:
removed

Reply:
it is not removed :(

Reply:
Oh sorry, I replied you incorrectly. This unwrap is actually guaranteed to succeed because the returned dictionary will always have `shard_id` in there

Reply:
ah I see. Let's add a comment here